From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Wed, 19 Jun 2024 15:59:10 -0600
Subject: [PATCH] dcomp: HACK: Try to workaround Chromium race fixed by
 Chromium commit 1c13c9105b40e8.

CW-Bug-Id: #23945
Valve-commit: 4dcdf9c15dc7743b1f1652be99c1b87ce39d49f8
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/dcomp/device.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/dlls/dcomp/device.c b/dlls/dcomp/device.c
index 2744d75..d0eb920 100644
--- a/dlls/dcomp/device.c
+++ b/dlls/dcomp/device.c
@@ -36,6 +36,9 @@ HRESULT WINAPI DCompositionCreateDevice2(IUnknown *rendering_device, REFIID iid,
 {
     FIXME("%p, %s, %p.\n", rendering_device, debugstr_guid(iid), device);
 
+    /* Try to workaround Chromium race fixed by Chromium commit 1c13c9105b40e83f87ebcad104fab57fcef43eb9. */
+    Sleep(50);
+
     return E_NOTIMPL;
 }
 
